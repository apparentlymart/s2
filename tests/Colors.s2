layerinfo type = core;
layerinfo "majorversion" = 0;

class Color
{
  var int r;
  var int g;
  var int b;
  function toString() : string;
}

function builtin string(int i) : string;

function Color::toString : string
{
  return "rgb(" + 
    string($.r) + "," +
    string($.g) + "," +
    string($.b) + ")";
}

function main()
{
  var Color c = new Color;
  $c.r = 100;
  $c.g = 0;
  $c.b = 50;

  "$c\n";
  "My color is $c\n";
}
