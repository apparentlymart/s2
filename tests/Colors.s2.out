rgb(100,0,50)
My color is rgb(100,0,50)
